Automattic/jetpack
# Server Sandbox: Move the server sandbox to the connection package

#### Changes proposed in this Pull Request:
The server sandbox functionality should be available to all plugins that use the connection package. So, let's move the server sandbox from the Jetpack plugin to the connection package.

 * Move the server sandbox functions from the Jetpack plugin to a new `Server_Sandbox` class in the connection package.
 * Deprecate the server sandbox functions in the Jetpack plugin.
 * Add unit tests for the new `Sandbox_Server` class.
 * Initialize the server sandbox in the connection package's `Manager` class.
 * Add a new item to the top admin bar which indicates that the Jetpack API is sandboxed.

#### Jetpack product discussion
* n/a

#### Does this pull request change what data or activity we track or use?
* no

#### Testing instructions:

##### Test site setup
1. Make sure that the `WP_DEBUG` constant is set to true.
2. Turn on your sandbox.

##### Test with Jetpack
3. Install and activate this branch of Jetpack.
4. Set the `JETPACK__SANDBOX_DOMAIN` constant to your sandbox.
5. Connect.
6. Check the `debug.log` file to verify that logs for for sandboxing have been generated.
7. Verify that the top admin bar displays a `Jetpack API sandboxed` menu item, and that hovering over that menu item displays your sandbox.
8. Disconnect Jetpack.
9. Remove the `JETPACK__SANDBOX_DOMAIN` constant.
10. Connect.
11. Check the `debug.log` file to verify that logs for for sandboxing have not been generated during the connection process in step 10.
12. Verify that the top admin bar does not display the `Jetpack API sandboxed` menu item.
13. Deactivate Jetpack.

##### Test with another plugin (Boost or Backup)
14. Install and activate this branch of another connected monorepo plugin (Boost or Backup).
15. Repeat steps 4 through 7.
16. Remove the `JETPACK__SANDBOX_DOMAIN` constant.
17.  Check the `debug.log` file to verify that logs for for sandboxing are no longer generated.
18. Verify that the top admin bar does not display the `Jetpack API sandboxed` menu item.
